staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -264,7 +264,10 @@ func (g *Cloud) InstanceExists(ctx context.Context, node *v1.Node) (bool, error)
 	providerID := node.Spec.ProviderID
 	if providerID == "" {
 		var err error
-		if providerID, err = g.InstanceID(ctx, types.NodeName(node.Name)); err != nil {
+		if providerID, err = cloudprovider.GetInstanceProviderID(ctx, g, types.NodeName(node.Name)); err != nil {
```

Comment:
This looks fine to me, but given we merged with this bug, this deserves some unit test.

Can you please add one?

Reply:
Done, @wojtek-t PTAL.

Reply:
Done, PTAL.